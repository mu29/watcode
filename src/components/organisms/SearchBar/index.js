import React from 'react'
import styled from 'styled-components'
import { Field } from 'redux-form'
import { ReduxField, IconButton } from 'components'

const Wrapper = styled.div`
  display: flex;
  align-items: flex-end;
  width: 40%;
  min-width: 22rem;
  padding: 0.25rem;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.1), 0 0 0 0.0625rem rgba(0,0,0,0.08);
  transition: box-shadow 0.2s;

  &:focus-within {
    box-shadow: 0 0.25rem 0.25rem 0 rgba(0,0,0,0.1), 0 0 0 0.0625rem rgba(0,0,0,0.08);
  }
`

const StyledField = styled(props => <Field { ...props } />)`
  min-width: 18.5rem;
  border: none;
  border-radius: 0;
  margin-bottom: 0;

  &:active, &:focus {
    border: none;
  }
`

const StyledIconButton = styled(props => <IconButton { ...props } />)`
  padding: 0 1rem;
  border-radius: 0;
`

const SearchBar = () => (
  <Wrapper>
    <StyledField name="query" placeholder="무엇을 찾으세요? (번호, 제목, 작가, 태그)" component={ ReduxField } />
    <StyledIconButton backgroundColor="white.default" color="yellow.default" icon="search" />
  </Wrapper>
)

export default SearchBar
