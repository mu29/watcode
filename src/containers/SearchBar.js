import React from 'react'
import { reduxForm } from 'redux-form'
import { SearchBar } from 'components'

const SearchBarContainer = props => <SearchBar { ...props } />

const onSubmit = (data, dispatch) => {}

export default reduxForm({
  form: 'search',
  onSubmit,
})(SearchBarContainer)
