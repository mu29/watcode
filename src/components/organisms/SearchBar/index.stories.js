import React from 'react'
import { storiesOf } from '@storybook/react'
import { SearchBar } from 'containers'

storiesOf('Organisms/SearchBar', module)
  .add('기본', () => (
    <SearchBar location="/" />
  ))
