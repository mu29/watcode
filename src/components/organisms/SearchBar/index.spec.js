import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import SearchBar from '.'

const wrap = (props = {}) => shallow(<SearchBar { ...props } />)

describe('<SearchBar />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })
})
