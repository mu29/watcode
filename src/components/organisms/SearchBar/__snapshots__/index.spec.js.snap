// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> 정상적으로 렌더링된다 1`] = `
<Wrapper>
  <StyledInput
    name="query"
    placeholder="번호, 제목, 작가, 태그로 검색하세요"
  />
  <StyledIconButton
    backgroundColor="white.default"
    color="yellow.default"
    icon="search"
  />
</Wrapper>
`;
